Assets/MixedRealityToolkit/_Core/DataProviders/SpatialObservers/WindowsMixedReality/WindowsMixedRealitySpatialObserver.cs

```diff
@@ -121,33 +104,19 @@ public override void Destroy()
         /// one as a spare for use in outstanding mesh requests. That way, we'll have fewer
         /// game object create/destroy cycles, which should help performance.
         /// </summary>
-        protected SpatialMeshObject? spareMeshObject = null;
+        private SpatialMeshObject? spareMeshObject = null;
 
         /// <summary>
         /// The time at which the surface observer was last asked for updated data.
         /// </summary>
         private float lastUpdated = 0;
-#endif // UNITY_WSA
 
         /// <inheritdoc />
-        public override IDictionary<int, GameObject> Meshes
-        {
-            get
-            {
-                Dictionary<int, GameObject> meshes = new Dictionary<int, GameObject>();
-                // NOTE: We use foreach here since Dictionary<key, value>.Values is an IEnumerable.
-                foreach (int id in meshObjects.Keys)
-                {
-                    meshes.Add(id, meshObjects[id].GameObject);
-                }
-                return meshes;
-            }
-        }
+        public override IReadOnlyDictionary<int, SpatialMeshObject> Meshes => SpatialMeshObjects;
```

Comment:
unfortunately, this pattern will not protect the caller from changes to the underlying collection.  the following snippet demonstrates the exception that occurs (InvalidOperation, collection was modified).

```
        Dictionary<int, float> dataSet = new Dictionary<int, float>();
        IReadOnlyDictionary<int, float> ReadOnlyDataSet => dataSet;

            InitializeDataSet();

            Random random = new Random(DateTime.Now.Second);

            while (true)
            {
                foreach(int i in ReadOnlyDataSet.Keys)
                {
                    // Print the key-value pair
                    Console.WriteLine($"{i} : {ReadOnlyDataSet[i]}");

                    // Remove the item
                    Console.WriteLine($"Remove {i}");
                    dataSet.Remove(i);

                    // Add a new item to the dictionary
                    Console.WriteLine("Add new item");
                    dataSet.Add(random.Next(), (float)random.NextDouble());
                }
            }
```

Reply:
I have a fix for the readonly / copy requirement that I will submit as a PR on @StephenHodgson's fork.